### PR TITLE
AP_Avoidance: Change division to multiplication and summarize processing

### PR DESCRIPTION
**Pull request by muramura**
1. I saw some operations dividing and multiplying floating values.
In STM32, multiplication is faster than division.
I change division to multiplication.

2. I think that if two if statements return the same value, it is better to use one if statement.

3. After setting the current time when the timeout detection variable value is 0, the condition is not satisfied for the next if statement, so that unnecessary processing is prevented.

### CODE REVIEW

File: libraries/AP_Avoidance/AP_Avoidance.cpp [hunk 1]
```diff
@@ -384,10 +384,7 @@ void AP_Avoidance::update_threat_level(const Location &my_loc,
 }
 
 MAV_COLLISION_THREAT_LEVEL AP_Avoidance::current_threat_level() const {
-    if (_obstacles == nullptr) {
-        return MAV_COLLISION_THREAT_LEVEL_NONE;
-    }
-    if (_current_most_serious_threat == -1) {
+    if (_obstacles == nullptr || _current_most_serious_threat == -1) {
```
Comment (peterbarker) [inline libraries/AP_Avoidance/AP_Avoidance.cpp]: Please don't coalesce these.  I have mentioned this before.  It all starts off great - until nobody can read the if-clause.

File: libraries/AP_Avoidance/AP_Avoidance.cpp [hunk 2]
```diff
@@ -418,8 +415,7 @@ void AP_Avoidance::handle_threat_gcs_notify(AP_Avoidance::Obstacle *threat)
         // only send cleared messages for a few seconds:
         if (_gcs_cleared_messages_first_sent == 0) {
             _gcs_cleared_messages_first_sent = now;
-        }
-        if (now - _gcs_cleared_messages_first_sent > _gcs_cleared_messages_duration * 1000) {
+        } else if (now - _gcs_cleared_messages_first_sent > _gcs_cleared_messages_duration * 1000) {
```
Comment (khancyr) [inline libraries/AP_Avoidance/AP_Avoidance.cpp]: This is good but should have been in its own commit

File: libraries/AP_Avoidance/AP_Avoidance.cpp (hunk 1 above)
Comment (muramura) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: @peterbarker san..
I'm coding with sequence diagrams and flowchart diagrams. Writing the same process multiple times is not good code. There may be many conditions.

Comment (muramura) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: @peterbarker san..
I'm coding with sequence diagrams and flowchart diagrams. Writing the same process multiple times is not good code. There may be many conditions.

Comment (magicrub) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: I'm in favor of grouping them

File: libraries/AP_Avoidance/AP_Avoidance.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/AP_Avoidance/AP_Avoidance.cpp]: agreed. @muramura the secret to getting PRs merged quickly is to make sure they are "obviously correct". Multiple PRs are better than fix-creep like this

**Comment by magicrub** [comment]
if it wasn't for that "else if" change, I wonder if this changes the binary at all? I'm curious if gcc automatically optimizes the const div to const mult.

**Comment by peterbarker** [comment]
Check this method out: https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Mission/AP_Mission.cpp#L1939

That entire routine could be done as a single return statement.

At the moment it's entirely readable.  You could fix that.

**Comment by khancyr** [comment]
Closing as the changes requested aren't done and the gain are very little if any ... Feel free to reopen with the change done. Thanks

**Comment by magicrub** [comment]
I really wish those div to mult changes can still get in.

**Pull request closed without merging**